src/com/t_oster/liblasercut/BlackWhiteRaster.java

```diff
@@ -89,77 +88,128 @@ public BlackWhiteRaster(GreyscaleRaster src, DitherAlgorithm dither_algorithm) t
   {
     this(src, dither_algorithm, null);
   }
-  
+
   public BlackWhiteRaster(GreyscaleRaster src, DitheringAlgorithm alg) throws InterruptedException
   {
     this(src, alg, null);
   }
 
-  public BlackWhiteRaster(int width, int height, byte[][] raster)
+  public BlackWhiteRaster(int width, int height)
   {
-    this.width = width;
-    this.height = height;
-    this.raster = raster;
+    this(width, height, 1, 1);
   }
 
-  public BlackWhiteRaster(int width, int height)
+  public BlackWhiteRaster(int width, int height, int bitDepth)
   {
-    this.width = width;
-    this.height = height;
-    this.raster = new byte[(width + 7) / 8][height];
+    this(width, height, bitDepth, 1);
   }
 
-  public boolean isBlack(int x, int y)
+  public BlackWhiteRaster(int width, int height, int bitDepth, int samplesPerPixel)
   {
-    int bx = x / 8;
-    int ix = 7 - (x % 8);
-    return ((raster[bx][y] & 0xFF) & (int) Math.pow(2,ix)) != 0;
+    this.raster = new RasterElement(width, height, bitDepth, samplesPerPixel);
   }
 
-  public void setBlack(int x, int y, boolean black)
+  public int getPixel(int x, int y)
```

Comment:
Could you add some method descriptions what exactly getPixel and setPixel do?

Reply:
Do note setBlack just got moved, it's down on line 168
`public void setBlack(int x, int y, boolean black)`

They delegate to the RasterElement which literally just sets the pixel. I mean I guess it sets the rightmost bitDepth bits of the integer into the given to the pixel space regardless of byte boundaries or other limitations. I could certainly do a better job documenting that.

But, I think including the delegate there was in error. The original BlackWhiteRaster didn't have a set pixel so this one might be better off without it. Though clearly, nothing is currently calling that code.

With the typical bitDepth of 1 this will simply set the pixel to 0 or 1 equal to (pixel & 1). also, getPixel (x,y) will return 0 or 1 depending on what that pixel's state is. It's basically the same as isBlack(x,y) at that point except that it doesn't decree that 1 is a black and 0 is white. It just gives the pixel's actual state. The function it's delegating to does this same thing for all bitDepths  (n <=32). Here the raster should be a 1 bit raster and thus just do what's expected for bitDepth 1. Namely set a 1 or 0 to the pixel in question or get a 1 or 0 from the pixel requested. I'll just add some documentation there. Shouldn't need much, it's called setPixel() and getPixel() and it sets and gets a pixel.